### PR TITLE
plugin/less-pretty-cat improvements

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
Adopt `_command_exists`, quote variables, handle unbound parameters, code flow simplification.

## Motivation and Context
This was part of my `_command_exists` branch (#1938) but was out of scope.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:25]: The bash-it standard is just `param '1: ...` so we can lose the `$` and the `shellcheck disable`

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:18]: I see the goal here, but *how* could `pygmetize` fail in a way that would defer to `cat` ?

Even if it couldn't offer any colorization, seems it would still act as a standard `cat` in that case?

The `cless` function assumes no failures, which further enforces the idea.

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:12]: Shouldn't the example be `ccat` ?

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:11]: I couldn't find any other use of `*` in params within bash-it and my search-fu failed me - I don't have a specific issue with it, but wanted to confirm it was standard, but could not confirm ...

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:26]: Shouldn't the example be `cless` ?

**Review by davidpfarrell — CHANGES_REQUESTED** [review]
Greetings all !

I have a few comments, many related to the core logic and not necessarily this PR

However : WHY ARE THERE 7 EMPTY FILES IN THIS PR?

If this is chained and waiting on another PR, let's mark it as a DRAFT until its ready.

File: plugins/available/less-pretty-cat.plugin.bash
Comment (gaelicWizard) [reply plugins/available/less-pretty-cat.plugin.bash:11]: I have no idea

File: plugins/available/less-pretty-cat.plugin.bash
Comment (gaelicWizard) [reply plugins/available/less-pretty-cat.plugin.bash]: No idea; I've removed that bit now

**Comment by gaelicWizard** [comment]
Rebased on master

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [reply plugins/available/less-pretty-cat.plugin.bash:11]: Yea me either ... I'm okay with it, we'll have to see how the generated docs look and if there are any issues with them.

Assuming not, I say its now our standard for all-arg parms :)

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:10]: Let's update this commend since the `or cat` isn't relevant any longer.

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:19]: With the `||| cat` removed, wondering if we can replace this for loop with just : 
```
pygmentize -f 256 -O style="$BASH_IT_CCAT_STYLE" -g "$@"
```
which is essentially the `cless` logic without the less call ...

#whats_good_for_cless_is_good_for_ccat

**Review by davidpfarrell — CHANGES_REQUESTED** [review]
Looking good I think we're almost there !

File: plugins/available/less-pretty-cat.plugin.bash
Comment (davidpfarrell) [inline plugins/available/less-pretty-cat.plugin.bash:14]: I think we can lose this now ...

**Review by davidpfarrell — APPROVED** [review]
Lets move this one along ...

one last note: On `cless`, although the docs mention a single file, the code actually cats ALL files and _then_ pipes the result to less.  I'm not going to hold up the PR on that but just wanted to call it out ...

**Comment by gaelicWizard** [comment]
> Lets move this one along ...
> 
> one last note: On `cless`, although the docs mention a single file, the code actually cats ALL files and _then_ pipes the result to less. I'm not going to hold up the PR on that but just wanted to call it out ...

tweaked and pushed 👍

**Pull request merged**
